compiler/qsc_parse/src/prim.rs

```diff
@@ -107,15 +89,17 @@ pub(super) fn path(s: &mut ParserContext) -> Result<Box<Path>> {
 
     let name = parts.pop().expect("path should have at least one part");
     let namespace = match (parts.first(), parts.last()) {
-        (Some(first), Some(last)) => {
-            let lo = first.span.lo;
-            let hi = last.span.hi;
-            Some(Box::new(Ident {
-                id: NodeId::default(),
-                span: Span { lo, hi },
-                name: join(parts.iter().map(|i| &i.name), ".").into(),
-            }))
-        }
+        (Some(_), Some(_)) => Some(
```

Comment:
I'm not sure doing `(parts.first(), parts.last())` has the same purpose anymore. I think it would be clearer to have `parts.is_empty()`.